CountItems/utils/utils.js

```diff
@@ -9,79 +9,111 @@ function consolidateDataMetrics(target, source) {
-    resTarget.objectCount._nonCurrentRestored += objectCount && objectCount._nonCurrentRestored ? objectCount._nonCurrentRestored : 0;
-    resTarget.objectCount._incompleteMPUUploads += objectCount && objectCount._incompleteMPUUploads ? objectCount._incompleteMPUUploads : 0;
+    resTarget.objectCount.current += BigInt(objectCount?.current || 0n);
+    resTarget.objectCount.nonCurrent += BigInt(objectCount?.nonCurrent || 0n);
+    resTarget.objectCount.deleteMarker += BigInt(objectCount?.deleteMarker || 0n);
+    resTarget.objectCount._currentCold += BigInt(objectCount?._currentCold || 0n);
+    resTarget.objectCount._nonCurrentCold += BigInt(objectCount?._nonCurrentCold || 0n);
+    resTarget.objectCount._currentRestoring += BigInt(objectCount?._currentRestoring || 0n);
+    resTarget.objectCount._currentRestored += BigInt(objectCount?._currentRestored || 0n);
+    resTarget.objectCount._nonCurrentRestoring += BigInt(objectCount?._nonCurrentRestoring || 0n);
+    resTarget.objectCount._nonCurrentRestored += BigInt(objectCount?._nonCurrentRestored || 0n);
+    resTarget.objectCount._incompleteMPUUploads += BigInt(objectCount?._incompleteMPUUploads || 0n);
 
-    resTarget.usedCapacity._inflightsPreScan += usedCapacity && usedCapacity._inflightsPreScan ? usedCapacity._inflightsPreScan : 0;
+    resTarget.usedCapacity._inflightsPreScan += BigInt(usedCapacity?._inflightsPreScan || 0n);
     if (accountOwnerID) {
         resTarget.accountOwnerID = accountOwnerID;
     }
 
     resTarget.usedCapacity.current += usedCapacity
-        ? usedCapacity._currentCold + usedCapacity._currentRestored + usedCapacity._currentRestoring
-        + usedCapacity._incompleteMPUParts : 0;
+        ? BigInt(usedCapacity._currentCold) + BigInt(usedCapacity._currentRestored) + BigInt(usedCapacity._currentRestoring)
+        + BigInt(usedCapacity._incompleteMPUParts) : 0n;
     resTarget.usedCapacity.nonCurrent += usedCapacity
-        ? usedCapacity._nonCurrentCold + usedCapacity._nonCurrentRestored + usedCapacity._nonCurrentRestoring : 0;
+        ? BigInt(usedCapacity._nonCurrentCold) + BigInt(usedCapacity._nonCurrentRestored) + BigInt(usedCapacity._nonCurrentRestoring) : 0n;
     resTarget.objectCount.current += objectCount
-        ? objectCount._currentCold + objectCount._currentRestored + objectCount._currentRestoring
-        + objectCount._incompleteMPUUploads : 0;
+        ? BigInt(objectCount._currentCold) + BigInt(objectCount._currentRestored) + BigInt(objectCount._currentRestoring)
+        + BigInt(objectCount._incompleteMPUUploads) : 0n;
     resTarget.objectCount.nonCurrent += objectCount
-        ? objectCount._nonCurrentCold + objectCount._nonCurrentRestored + objectCount._nonCurrentRestoring : 0;
+        ? BigInt(objectCount._nonCurrentCold) + BigInt(objectCount._nonCurrentRestored) + BigInt(objectCount._nonCurrentRestoring) : 0n;
 
     return resTarget;
 }
 
+
+function serializeBigInts(obj) {
```

Comment:
nit: performance-wise, it is better to have such generic functions or custom functions for the specific objects we convert? (since we it is actually used for a know type `BucketInfo`) ?

Reply:
As discussed I believe this is minor as ran once per bucket, and eventually it allows us to be flexible in the future, if we were to add more fields.